TuneContainer

DebarshiB13 opened the pull request:
### Ticket Link

---

### Related Links

---

### Description
Generated TuneContainer to show data fetched from itunes api. Added tests and code for TuneContainer Saga, Reducer, & Api.
---

### Steps to Reproduce / Test

---
 * Go to localhost:3000/songs
 * Open the Redux Dev tool.
 * Type any song or artist name on the search bar, see the tuneContainer state change based on input and response .
### GIF's

---
![action](https://user-images.githubusercontent.com/89914287/132009663-cc76b5a8-c2fe-42d6-8e93-2abef78b1c48.gif)
![test](https://user-images.githubusercontent.com/89914287/132009734-6ea372eb-6edc-43b7-9590-e5f618cc74b6.gif)

Commit:
Generated TuneContainer

DebarshiB13 committed:
Added Tests for TuneContainer

Added code and tests for api service, index, saga, reducer for TuneContainer

DebarshiB13 committed:
Fixed Lint Issue

DebarshiB13 committed:
Commented SonarCloudScan in ci.yml

alichherawalla commented on app/containers/TuneContainer/index.js:
```diff
@@ -0,0 +1,128 @@
+  dispatchItuneSongs,
+  dispatchClearItuneSongs,
+  intl,
+  songsData = {},
```
this was a problem with the react-template too, i made the changes. 

use defaultProps instead of adding default values here

alichherawalla commented on app/containers/TuneContainer/reducer.js:
```diff
@@ -0,0 +1,39 @@
+        draft.songsError = null;
+        break;
+      case tuneContainerTypes.FAILURE_GET_ITUNE_SONGS:
+        draft.songsError = get(action.error, 'message', 'something_went_wrong');
```
draft.songsError = get(action.error, 'message', translate('something_went_wrong'));

DebarshiB13 replied:
Getting an error of **formatMessage** is undefined when the test runs as a result it fails. Seems like the value **intl** is not initialized.

DebarshiB13 replied:
Fixed the issue by creating **createintl** file in utils folder. Working as expected

alichherawalla commented on app/containers/TuneContainer/tests/index.test.js:
```diff
@@ -0,0 +1,49 @@
+  it('should render and match the snapshot', () => {
+    const { baseElement } = renderProvider(<TuneContainer dispatchItuneSongs={submitSpy} />);
+    expect(baseElement).toMatchSnapshot();
+  });
```
please add a test for if the `search-bar` is getting rendered

DebarshiB13 committed:
Minor Fix

Added test for SearchBar, removed default values from TuneContainer props

DebarshiB13 committed:
Minor fix 2

Fixed the message something went to wrong to match the exact translated message

DebarshiB13 committed:
Minor Fix 3

Rename file createintl to CreateIntl which fixed failed test.

DebarshiB13 committed:
Fixed issue with createintl module not found error
